pkg/kcp/provider/gcp/nuke/new.go

```diff
@@ -0,0 +1,43 @@
+package nuke
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	nuketypes "github.com/kyma-project/cloud-manager/pkg/kcp/nuke/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func New(stateFactory StateFactory) composed.Action {
+	return func(ctx context.Context, st composed.State) (error, context.Context) {
+
+		state, err := stateFactory.NewState(ctx, st.(nuketypes.State))
+		if err != nil {
+			obj := st.Obj().(*v1beta1.Nuke)
+			return composed.UpdateStatus(obj).
+				SetExclusiveConditions(metav1.Condition{
+					Type:    v1beta1.ConditionTypeError,
+					Status:  metav1.ConditionTrue,
+					Reason:  v1beta1.ReasonGcpError,
+					Message: err.Error(),
+				}).
+				SuccessError(composed.StopAndForget).
+				SuccessLogMsg(fmt.Sprintf("Error creating new GCP Nuke state: %s", err)).
```

Comment:
Cloud provider errors MUST not be reveled to users by putting them to the k8s resource status

Reply:
SuccessLogMessage doesn't put anything in the status. It only logs in info level. Anyway I removed it as I added a log message based on your previous comment and this was redundant.